### ISSUE TITLE
allow selecting all text between brackets

### ISSUE THREAD

**Issue by JeanMertz** [wez/wezterm#583]
Something Terminal.app do, that I find highly productive as a developer is that it allows you to select text between two brackets (including the brackets themselves) by double-clicking on the opening or closing bracket.

Given this line:

```
{ foo } bar
```

If I double-click `{` or `}` in Terminal.app, it'll select `{ foo }`.

In Wezterm, it'll select `{ ` (with space after) or ` }` (with space before).

It's worth mentioning that this works for multi-line input, nested brackets, and works equally well for parenthesis, braces, brackets, angle brackets, and possibly more.

Basically, this example works "the way you expect it to work":

```
{ \
    foo \
    } \
        \
        { foo { bar } \
    } { foo [ bar ] } < > < foo >
```
